docs/source/typedd/typedd.rst

```diff
@@ -281,28 +281,23 @@ in the type:
 .. code-block:: idris
 
     snocListHelp : {input : _} ->
-                   (snoc : SnocList input) -> (rest : List a) -> SnocList (input +
+                   (snoc : SnocList input) -> (rest : List a) -> SnocList (input ++ rest)
 
 It's no longer necessary to give ``{input}`` explicitly in the patterns for
 ``snocListHelp``, although it's harmless to do so.
 
-In ``IsSuffix.idr``, the matching has to be written slightly differently. The
-recursive with application in Idris 1 probably shouldn't have allowed this!
-Note that the ``Snoc`` - ``Snoc`` case has to be written first otherwise Idris
-generates a case tree splitting on ``input1`` and ``input2`` instead of the
-``SnocList`` objects and this leads to a lot of cases being detected as missing.
+``IsSuffix.idr`` also has to be adjusted to match the new ``Snoc`` arguments.
```

Comment:
The comment probably can be improved. I'm especially confused as to why the new code doesn't need a `| ysrec` at line 300.